src/App.js

```diff
@@ -1,25 +1,58 @@
-import React from 'react';
+import React, {Component} from 'react';
 import './App.css';
 import {Switch, Route, Redirect} from 'react-router-dom';
 import Posts from "./containers/Posts/Posts";
 import AddPost from "./containers/AddPost/AddPost";
 import Auth from "./containers/Auth/Auth";
+import Logout from "./containers/Auth/Logout/Logout";
+import {connect} from "react-redux";
+import * as actions from './store/actions/index';
 
-let routes = (
-    <Switch>
-      <Route path={"/posts"} component={Posts}/>
-      <Route path={"/add-post"} component={AddPost}/>
-      <Route path={"/auth"} component={Auth}/>
-      <Redirect to={'/auth'} from={'/'}/>
-    </Switch>
-)
-
-function App() {
-  return (
-    <div className="App">
-      {routes}
-    </div>
-  );
+
+
+class App extends Component{
+    componentDidMount() {
+        this.props.tryAutoSignup();
+    }
+
+    render() {
+        let routes = (
+            <Switch>
+                <Route path={"/auth"} component={Auth}/>
+                <Redirect to={'/auth'} from={'/'}/>
+            </Switch>
+        )
+
+        if(this.props.isAuthenticated){
+            routes = (
+                <Switch>
+                    <Route path={"/posts"} component={Posts}/>
+                    <Route path={"/add-post"} component={AddPost}/>
+                    <Route path={'/logout'} component={Logout}/>
+                    <Route path={"/auth"} component={Auth}/>
+                    <Redirect to={'/posts'} from={'/'}/>
+                </Switch>
+            )
+        }
```

Comment:
Another approach to handle such case is using private routes 
Just for knowledge 
https://medium.com/@tomlarge/private-routes-with-react-router-dom-28e9f40c7146